src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1631,6 +1631,191 @@ case class BinaryOp(posn: Position, lhs: AST, operation: String, rhs: AST) exten
       s.r.findFirstIn(t).isDefined
     }
 
+    case ("+", TArray(elementType)) =>
+      ((lhs.`type`, rhs.`type`, elementType): @unchecked) match {
+        case (t: TNumeric, TArray(_), TDouble) => AST.evalCompose[Any, IndexedSeq[_]](ec, lhs, rhs) { case (num, arr) =>
+          val d = DoubleNumericConversion.to(num)
+          arr.map(x => DoubleNumericConversion.to(x) + d)
+        }
```

Comment:
re: the above comments and some offline discussions, here's an update `liftArrayOp`:
```scala
  def liftArrayOp[T >: Null](coerce: Any => T, binOp: (T, T) => T): (Any, IndexedSeq[Any]) => IndexedSeq[T] =
    { case (num, arr) => {
      val y = coerce(num);
      arr.map(x => if (x == null) null else binOp(coerce(x), y))
    } }
```